hastory-server/src/Data/Hastory/Server/TestUtils.hs

```diff
@@ -56,7 +73,26 @@ withTestServer func = do
           void $ runSqlPool (runMigrationSilent migrateAll) siPool
           let mkApp = pure $ app opts settings
               opts = Options 10 Nothing
-              settings = ServerSettings siToken siPool
+              settings = ServerSettings siPool jwtSettings defaultCookieSettings
           testWithApplication mkApp $ \p ->
             let siClientEnv = mkClientEnv manager (BaseUrl Http "127.0.0.1" p "")
              in func (ServerInfo {..})
+
+type RegistrationData = (UserId, Token)
+
+withNewUser :: ClientEnv -> UserForm -> (RegistrationData -> Expectation) -> Expectation
+withNewUser clientEnv userForm func = do
+  Right userId <- runClientM (createUserClient userForm) clientEnv
+  Right resp <- runClientM (createSessionClient userForm) clientEnv
+  case extractJWTCookie resp of
+    Nothing -> expectationFailure "JWT Cookie not found"
+    Just cookie -> func (userId, cookie)
+
+createEntry :: ClientEnv -> Token -> SyncRequest -> IO (Either ClientError NoContent)
```

Comment:
usually you make a function like `runTestClient :: ClientEnv -> ClientM a -> IO a` so you don't have to make all of these.

Reply:
I found that doing something like `runTestClient` would be such a small wrapper around `runClientM` that I didn't think it worthwhile. `createEntry`, etc., were also very thin wrappers around `runClientM`, but they had the benefit of naming the function something significant. With a name as generic as `runTestClient`, I think it makes more sense to remove the functions.